operations/research-and-development/product/development-process/secure-software-development.md

```diff
@@ -0,0 +1,85 @@
+# Secure Software Development at Mattermost
+
+This document provides guidelines to secure software development at Mattermost. The document is at the moment a [1% draft](/company/about-mattermost/mindsets#drafts-at-1-50-99) and not conclusive.
+
+## Document scope
+
+This document covers the secure software development practices on all source repositories used to create software products offered by Mattermost.
+
+The product security team is overall in charge of secure software development practices by providing help and guidance and performing various verification activities, including code reviews and testing. Development teams are encouraged to adopt the secure software development practices described here. Third-party plugin developers are also welcome to use this document for guidance.
+
+## Recommended reading
+
+* [OSSF Scorecard Action Check Documentation](https://github.com/ossf/scorecard/blob/main/docs/checks.md)
+* [Overview of Software Supply Chain Attacks](https://www.crowdstrike.com/cybersecurity-101/cyberattacks/supply-chain-attacks/) by [Crowdstrike](https://www.crowdstrike.com/)
+* [OWASP Top Ten](https://owasp.org/www-project-top-ten/), a curated and updated list on most common security risks in web applications.
+* [OWASP Application Security Verification Standard](https://owasp.org/www-project-application-security-verification-standard/)
+* [OWASP Cheat Sheet Series](https://cheatsheetseries.owasp.org/)
+
+## Code reviews
+
+### Manual code reviews
+
+The product security team performs reviews on new features and plugins when necessary. Here's some of the things included in the checks:
+
+* Manual diff review
+* Static application security testing (SAST) using multiple tools
+* Software composition analysis (SCA)
+* Dynamic application security testing (DAST)
+
+Detailed review checklists are documented in the [security team shared drive](https://drive.google.com/drive/folders/1s3paBN6sfGRP-NHNuoekDHY_nrtRVagt), accessible to team members only.
+
+### Automated code reviews
+
+The key Mattermost repositories have automated code security analysis tools set up. If this is not the case, please reach out to the Product Security Team for assistance in setting this up. Particular automated tools in each repository depend on the technologies used. In practice, we use [CodeQL](https://codeql.github.com/) as a Github Action to automatically analyze all pull requests. Additionally, the use of the [OSSF Security Scorecards GitHub Action](https://github.com/ossf/scorecard) is strongly encouraged. This document addresses many of the best practices checked by the Scorecards action.
```

Comment:
```suggestion
Mattermost production repositories have automated code security analysis tools set up. If this is not the case for a specific repository, please reach out to the product security team for assistance.

The automated tools in each repository depend on the technologies used in the repository. In practice, we use [CodeQL](https://codeql.github.com/) as a Github Action to automatically analyze all pull requests. Additionally, the use of the [OSSF Security Scorecards GitHub Action](https://github.com/ossf/scorecard) is strongly encouraged. This document describes some of the best practices checked by the Scorecards action.
```

Change `key` to `production` as suggested by @DSchalla earlier. Plus minor clarifying changes in wording.